llvm/tools/spirv-to-ir-wrapper/CMakeLists.txt

```diff
@@ -1,14 +1,37 @@
-set(LLVM_LINK_COMPONENTS
-  BinaryFormat
-  Core
-  Support
-  SPIRVLib
+
+if(LLVM_LINK_LLVM_DYLIB)
+  set(LLVM_LINK_COMPONENTS
+    BinaryFormat
+    Core
+    Support
+  )
+else()
+  set(LLVM_LINK_COMPONENTS
+    BinaryFormat
+    Core
+    Support
+    SPIRVLib
   )
+endif()
+
```

Comment:
Please, remove duplication.
You can do it here by adding to the existing code.
```cmake
if (LLVM_LINK_LLVM_DYLIB)
  list (REMOVE_ITEM LLVM_LINK_COMPONENTS SPIRVLib)
endif()
```
Another option: create a list w/o `SPIRVLib` and add it if `LLVM_LINK_LLVM_DYLIB` is not set.

On the other hand, I expect `add_llvm_tool` to handle `LLVM_LINK_LLVM_DYLIB` correctly. Could you clarify what is the problem with `spirv-to-ir-wrapper` tool and `SPIRVLib` library?

Reply:
I expect GenXIntrinsics library to be linked statically to llvm.so library as all other libraries. Can we do that?

Reply:
Why did you [remove](https://github.com/intel/vc-intrinsics/commit/c456ff05bd8e11458772acef20e59ef18aa7876d) `GenXIntrinsics` library from `llvm.so` library? I think this is the reason of cycle dependency. All libraries in the error message are part of the llvm.so, except `LLVMGenXIntrinsics`, which also depends on LLVM libraries. That's why we have a conflict: `LLVMSYCLLowerIR` library (part of `llvm.so`) depends on `LLVMGenXIntrinsics` (standalone), which depends on `LLVMCore` library (part of `llvm.so`).
If you make `LLVMGenXIntrinsics` as I suggested above, there should be no cyclic dependencies as there will be just one `llvm.so` library.

Reply:
> cmake is more sensitive to circular dependencies when dynamic libraries are built

I don't get what it means.
Shouldn't LLVMGenXIntrinsics be a part of LLVMTarget library or some other LLVM target library? Isn't the root case of the problem is that we put intrinsics in a separate library as opposite to how all other intrinsics are handled?

Reply:
> Could you elaborate in the description the difference with regular intrinsics, which makes us customize cmake files for these tools, please?

GenX intrinsics come from an external repo during the build, unlike others, and they are built with the cmake files provided in the repo.